src/actor/Verifier.ts

```diff
@@ -0,0 +1,148 @@
+import * as gabi from '@kiltprotocol/portablegabi'
+import CType from '../ctype/CType'
+import {
+  IRequestClaimsForCTypes,
+  MessageBodyType,
+  ISubmitClaimsForCTypes,
+} from '../messaging/Message'
+import AttestedClaim from '../attestedclaim/AttestedClaim'
+import PublicAttesterIdentity from '../attesteridentity/PublicAttesterIdentity'
+
+/**
+ * A helper class to initiate a verification by creating a presentation request which is built
+ * on a specific [[CType]] and attributes of the [[Claim]] the verifier requires to see.
+ */
+export class PresentationRequestBuilder {
+  private builder: gabi.CombinedRequestBuilder
+  private ctypes: Array<CType['hash']>
+  constructor() {
+    this.builder = new gabi.CombinedRequestBuilder()
+    this.ctypes = []
+  }
+
+  /**
+   * Initiates a verification by creating a presentation request for a specific [[CType]].
+   * Note that you are required to call [[finalize]] on the request to conclude it.
+   *
+   * @param p The parameter object.
+   * @param p.ctypehash The SHA-256 hash of the [[CType]].
+   * @param p.attributes A list of attributes of the [[Credential]]s the Verifier has to see in order to verify it.
+   * @param p.legitimations An optional boolean representing whether the Verifier requests to see the legitimations of the Attesters which signed the [[Credential]]s.
+   * @param p.delegationId An optional boolean representing whether the Verifier requests to see the Attesters' unique delegation identifiers.
+   * @param p.reqUpdatedAfter The optional minimum required timestamp on which the [[Credential]] needs to be updated.
+   * The default value for this is the current date.
+   * @returns A [[PresentationRequestBuilder]] on which you need to call [[finalize]] to complete the presentation request.
+   */
+  public requestPresentationForCtype({
+    ctypeHash,
+    attributes,
+    legitimations,
+    delegationId,
+    reqUpdatedAfter = new Date(),
+  }: {
+    attributes: string[]
+    ctypeHash: CType['hash']
+    legitimations?: boolean
+    delegationId?: boolean
+    reqUpdatedAfter?: Date
+  }): PresentationRequestBuilder {
+    const rawAttribute = attributes.map(attr => `claim.contents.${attr}`)
```

Comment:
As we talked, we might want to refactor this, to only include `claim.contents.` for gabi

Reply:
We need to map this, because we are putting the property names in a PE request.